File: gr/babbling-brook.hs
-- babbling brook (jmcc) #SC3
-- http://lists.create.ucsb.edu/pipermail/sc-users/2007-April/033239.html

import Sound.SC3 {- hsc3 -}

babbling_brook :: UGen
babbling_brook =
  let b f m a g = let n3 = lpf (brownNoise 'α' AR) f * m + a
                      n4 = onePole (brownNoise 'β' AR) 0.99
                  in rhpf n4 n3 0.03 * g
      x = uclone 'γ' 2 (b 14 400 500 0.006)
      y = uclone 'δ' 2 (b 20 800 1000 0.010)
  in x + y

main :: IO ()
main = audition (out 0 babbling_brook)

File: gr/babbling-brook.fs
\ babbling brook (jmcc) #SC3

: b { f m a g } BrownNoise.ar 0.99 OnePole BrownNoise.ar f LPF m * a + 0.03 RHPF g * ;
14 400 500 0.006 b 2 clone 20 800 1000 0.01 b 2 clone + play

